addons/binding/org.openhab.binding.boschindego/src/main/java/org/openhab/binding/boschindego/handler/BoschIndegoHandler.java

```diff
@@ -0,0 +1,249 @@
+            updateState(READY, new DecimalType(ready ? 1 : 0));
+            updateState(ERRORCODE, new DecimalType(error));
+            updateState(MOWED, new PercentType(mowed));
+            updateState(STATE, new DecimalType(eshStatus));
+            updateState(TEXTUAL_STATE, new StringType(statusWithMessage.getMessage()));
+
+        } catch (IndegoAuthenticationException e) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR);
+        } catch (IndegoException e) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR);
+        }
+    }
+
+    private boolean isReadyToMow(int statusCode, int error) {
+        return (statusCode == 258 || statusCode == 260 || statusCode == 261 || statusCode == 517 || statusCode == 519)
+                && error == 0;
+    }
+
+    private boolean verifyCommand(int command, int eshStatus, int statusCode, int errorCode) {
+        // Mower reported an error
+        if (errorCode != 0) {
+            logger.error("The mower reported an error.");
+            return false;
+        }
+        // Command out of range
+        if (command < 1 || command > 3) {
+            logger.debug("Command out of range");
+            return false;
+        }
+        // Command is equal to current state
+        if (command == eshStatus) {
+            logger.debug("Command is equal to state");
+            return false;
+        }
+        // Cant pause while the mower is docked
+        if (command == 3 && eshStatus == 2) {
+            logger.debug("Can´t pause the mower while it´s docked or docking");
+            return false;
+        }
+        // Command means "MOW" but mower is not ready
+        if (command == 1 && !isReadyToMow(statusCode, errorCode)) {
+            logger.debug("The mower is not ready to mow in the moment");
+            return false;
+        }
+        return true;
+    }
+
+    private DeviceCommand getCommandFromEshStatus(int eshStatus) {
+        DeviceCommand command;
+        switch (eshStatus) {
+            case 1:
+                command = DeviceCommand.MOW;
+                break;
+            case 2:
+                command = DeviceCommand.RETURN;
+                break;
+            case 3:
+                command = DeviceCommand.PAUSE;
+                break;
+            default:
+                command = null;
+        }
+        return command;
+    }
+
+    private int getEshStatusFromCommand(DeviceCommand command) {
+        int eshStatus;
+        switch (command) {
+            case MOW:
+                eshStatus = 1;
+                break;
+            case RETURN:
+                eshStatus = 2;
+                break;
+            case PAUSE:
+                eshStatus = 3;
+                break;
+            default:
+                eshStatus = 0;
+        }
+        return eshStatus;
+    }
+
+    @Override
+    public void handleRemoval() {
+        super.handleRemoval();
```

Comment:
```ThingHandler.dispose()``` is what you were looking for to override. You may consider it as the symmetric pendant to ```initialize()```

FTR: ```ThingHandler.handleRemoval()``` is only called when a Thing _really_ is removed by the user. But not in case of shutdown etc.